program/src/main/System.vala

```diff
@@ -125,12 +136,48 @@ public class System
 
             
             //right side: marks + cgi + hotkeys
+            var util_pane = new Gtk.Paned(Gtk.Orientation.VERTICAL);
+
+            //put marksGrid in scrollable in case of overflow
+            var scroll = new Gtk.ScrolledWindow(null,null);
+            scroll.set_size_request(230,500);
+            scroll.set_propagate_natural_width(true);
+
             marksGrid = new Gtk.Grid();
             marksGrid.set_row_spacing(5);
             marksGrid.set_column_spacing(5);
-            marksGrid.set_size_request(100, -1); //100 = minimum width
-            hpaned.pack2(marksGrid, false, false); //put the vpaned on the right side of the hpaned
+            marksGrid.set_size_request(230, 500); //100 = minimum width
+
```

Comment:
this is a lot of code. It should be put in a function and called. That might even be able to be broken into more functions

Reply:
I definitely agree it's a lot of code and takes way too long to scroll through, but it's still within the group of window area UI code during init and makes sense there imo. 

Do you think it's worth having functions like `UI_element_name_setup()` for all the UI elements and call that one by one just to break it up? If so, we should probably throw that in a new branch and do the same for the other UI element code segments as well (menuBar, submenus, etc. etc.)